### ISSUE TITLE
Port Image_pipeline on ROS2

### ISSUE THREAD

**Issue by yechun1** [ros-perception/image_pipeline#360]
Image_pioeline porting status: (updated on 7/8/2019)

- [x] [depth_image_proc] - **Merged (PR: #362)**
- [x] [image_publisher] - **Merged(PR: #366 )**
- [x] [image_rotate] -  **Merged(PR:https://github.com/ros-perception/image_pipeline/pull/385)**
- [x] [image_proc] - **Merged(PR: ~#369~  https://github.com/ros-perception/image_pipeline/pull/426)**
- [x] [image_view] - **Merged(PR: ~#475~ #360)**
- [x] [camera_calibration] - **in review (PR: ~#376~ #447)**
- [x] [stereo_image_proc] - no owner, depends on image_proc **in review (PR: #486)**

**Comment by yechun1** [comment]
Could image_pipeline's owner/maintainer please help to create ros2 branch for [ros-perception/image_pipeline](https://github.com/ros-perception/image_pipeline)?

**Comment by yechun1** [comment]
@mjcarroll thanks for help to create ros2 branch, first PR to enable ros2 depth_image_proc is ready on #362.

**Comment by jacobperron** [comment]
Any update on `stereo_image_proc`?

**Comment by SteveMacenski** [comment]
I can tell you @JWhitleyAStuff and I aren't thinking about it. I had alot of the ROS2 work in this repo scheduled post-ROSCon as I'm a little preoccupied with Navigation2 and slam toolbox. 

@jacobperron is that something you're motivated to look at?

**Comment by yechun1** [comment]
I tried to port stereo_image_proc in the beginning of the year but blocked by image_proc which is not ready then. And now the image_proc has been ported and stereo_image_proc porting could be continue, but very sorry that I have no resource on this right now. 
It would be glad if some one could have a time to work on it. Below is previously patch(not fully completed) just for reference.

port ros2 stereo_image_proc: https://github.com/yechun1/image_pipeline/commit/e9993bed6220d28488fa73516eb95110d68f804e
on (https://github.com/yechun1/image_pipeline/tree/ros2_stereo_image_proc)

**Comment by klintan** [comment]
@yechun1 one version of image_proc is actually already merged, unfortunately it is not possible to run it as a component yet, but I'll put in some PRs for that. However I think it might still be considered "ready/done" for image_pipeline. (at least as a initial release candidate) 

edit: and the PR you are referencing for image_proc is merged and closed :)

**Comment by jacobperron** [comment]
> @jacobperron is that something you're motivated to look at?

@SteveMacenski I don't have the bandwidth at the moment, I'll let you know if things change.

**Comment by SteveMacenski** [comment]
@jacobperron that doesn't surprise me in the slightest :wink:

**Comment by jacobperron** [comment]
FYI, I have some bandwidth now and will start porting `stereo_image_proc`.

**Comment by laszloturanyi** [comment]
Hi,
I managed to port stereo_image_proc some days ago. There is still some work to do to pass all the tests but now it can function.

@jacobperron you do not have to do it from scratch I rebased it to your jacob/ros2_image_processor
you can find a link to my fork. (https://github.com/laszloturanyi/image_pipeline/tree/ros2_stereo_image_proc)

I would be glad if you could have a look at it I have limited capacity to finish this but maybe we can find a way to do this efficiently.

Also some other notes:
- .cfg files should removed (not only in stereo_image_proc) after comments are copied and parameter limits are handled correctly.
- launch file is still the old one from ROS1
- QoS policy is not consistent in image_pipeline. e.g disparity output of depth_image_proc will not be received by image_view nodes because the latter uses default profile while the first uses sensor data profile

**Comment by jacobperron** [comment]
@laszloturanyi Thanks for sharing your effort! I'm happy to build on your fork if you don't have the time to finish the port.

**Comment by laszloturanyi** [comment]
@jacobperron that would be great. I added you as collaborator.

**Comment by jacobperron** [comment]
See #486 for a port of stereo_image_proc.

@laszloturanyi I ended up not basing my fork on yours, as I had already done some of the same preliminary refactoring as you. It would be great if you could help review/try out the proposed port :)

I'll be adding support for parameter limits/reconfiguring shortly.
